### PR TITLE
Better structure POSIX API (inter)dependencies

### PR DESCRIPTION
**Pull request by pfalcon**
How it happens to be now, CONFIG_PTHREAD_IPC is the top-level master switch for POSIX APIs implemented in Zephyr (well, those which are implemented in lib/posix/, because e.g. Sockets API doesn't live there). This is confusing on its own (because "PTHREAD_IPC" doesn't mention POSIX), and cause any other API (e.g. filesystem) depend on pthreads. But pthreads itself is just a peer API to many others (time/clock, filesystem, etc.), and a niche one, so bringing it in for any other POSIX API doesn't make sense.

This PR introduces a dedicated master switch for POSIX APIs, CONFIG_POSIX_API. CONFIG_PTHREAD_IPC becomes just an API option under it, and other there are restructurings to make other APIs independent of it.

There're also other changes, like introduction of dirent.h, as required by POSIX to contain opendir() and friends.

I stumbled upon these issues while working on https://github.com/zephyrproject-rtos/zephyr/issues/7405 (literally, generic fdtable implementation would live in lib/posix/ , but that currently depends on pthreads, but I don't want to make sockets depend on pthreads!), so this is on critical path to it.

### CODE REVIEW

**Comment by pfalcon** [comment]
A couple more issues uncovered by CI, now fixed too, build passes.

**Review by nashif — APPROVED** [review]

File: tests/posix/fs/prj.conf [hunk 1]
```diff
@@ -1,6 +1,7 @@
 CONFIG_FILE_SYSTEM=y
 CONFIG_FAT_FILESYSTEM_ELM=y
 CONFIG_DISK_ACCESS_RAM=y
-CONFIG_PTHREAD_IPC=y
+CONFIG_POSIX_API=y
+#CONFIG_PTHREAD_IPC=y
```
Comment (nniranjhana) [inline tests/posix/fs/prj.conf]: You probably meant to remove this `CONFIG_PTHREAD_IPC` here

Comment (pfalcon) [reply tests/posix/fs/prj.conf]: @nniranjhana: Good catch, thought I squashed that out. Pushed update.

**Comment by pfalcon** [comment]
@nniranjhana : If it looks ok for you now, can you please approve it?

**Comment by nniranjhana** [comment]
@pfalcon Sure, changes LGTM

**Review by nniranjhana — APPROVED** [review]

**Pull request merged**